holoviews/streams.py

```diff
@@ -0,0 +1,195 @@
+"""
+The streams module defines the streams API that allows visualizations to
+generate and respond to events, originating either in Python on the
+server-side or in Javascript in the Jupyter notebook (client-side).
+"""
+
+import param
+import uuid
+from collections import OrderedDict
+
+
+
+class Preprocessor(param.Parameterized):
+    """
+    A Preprocessor is a callable that takes a dictionary as an argument
+    and returns a dictionary. Where possible, Preprocessors should have
+    valid reprs that can be evaluated.
+
+    Preprocessors are used to set the value of a stream based on the
+    parameter values. They may be used for debugging purposes or to
+    remap or repack parameter values before they are passed onto to the
+    subscribers.
+    """
+
+    def __call__(self, params):
+        return params
+
+
+
+class Rename(Preprocessor):
+    """
+    A preprocesor used to rename parameter values.
+    """
+
+    def __init__(self, **mapping):
+        self.mapping = mapping
+
+    def __call__(self, params):
+        return {self.mapping.get(k,k):v for (k,v) in params.items()}
+
+    def __repr__(self):
+        keywords = ','.join('%s=%r' % (k,v) for (k,v) in sorted(self.mapping.items()))
+        return 'Rename(%s)' % keywords
+
+
+
+class Stream(param.Parameterized):
+    """
+    A Stream is simply a parameterized object with parameters that
+    change over time in response to update events. Parameters are
+    updated via the update method.
+
+    Streams may have one or more subscribers which are callables passed
+    the parameter dictionary when the trigger classmethod is called.
+    """
+
+    # Mapping from uuid to stream instance
+    registry = OrderedDict()
+
+    @classmethod
+    def trigger(cls, streams):
+        """
+        Given a list of streams, collect all the stream parameters into
+        a dictionary and pass it to the union set of subscribers.
+
+        Passing multiple streams at once to trigger can be useful when a
+        subscriber may be set multiple times across streams but only
+        needs to be called once.
+        """
+        # Union of stream values
+        items = [stream.value.items() for stream in streams]
+        union = dict(kv for kvs in items for kv in kvs)
```

Comment:
Sure. I was planning to do that and then forgot.

Once issue is that ``self.warning`` seems to be showing up in the console and not the notebook...